src/System.Net.Security/src/System/Net/Security/SslApplicationProtocol.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+
+namespace System.Net.Security
+{
+    public struct SslApplicationProtocol : IEquatable<SslApplicationProtocol>
+    {
+        private readonly byte[] _protocol;
+        private readonly ReadOnlyMemory<byte> _readOnlyProtocol;
+        private static readonly Encoding s_utf8 = Encoding.GetEncoding(Encoding.UTF8.CodePage, EncoderFallback.ExceptionFallback, DecoderFallback.ExceptionFallback);
+
+        // h2
+        public static readonly SslApplicationProtocol Http2 = new SslApplicationProtocol(new byte[] { 0x68, 0x32 }, false);
+        // http/1.1
+        public static readonly SslApplicationProtocol Http11 = new SslApplicationProtocol(new byte[] { 0x68, 0x74, 0x74, 0x70, 0x2f, 0x31, 0x2e, 0x31 }, false);
+
+        internal SslApplicationProtocol(byte[] protocol, bool copy)
+        {
+            if (protocol == null)
+            {
+                throw new ArgumentNullException(nameof(protocol));
+            }
+
+            if (protocol.Length == 0)
+            {
+                throw new ArgumentException(SR.net_ssl_app_protocol_empty, nameof(protocol));
+            }
+
+            if (copy)
+            {
+                _protocol = new byte[protocol.Length];
+                Array.Copy(protocol, _protocol, protocol.Length);
+            }
+            else
+            {
+                _protocol = protocol;
+            }
+
+            _readOnlyProtocol = new ReadOnlyMemory<byte>(_protocol);
+        }
+
+        public SslApplicationProtocol(byte[] protocol) : this(protocol, true) { }
+
+        public SslApplicationProtocol(string protocol) : this(s_utf8.GetBytes(protocol), false) { }
```

Comment:
If this was the consensus I'll go along with it; but I (personally) would be happier if this ctor didn't exist.

Just sayin'.  Not putting up a fight over it.